entity-framework/core/what-is-new/ef-core-9.0/whatsnew.md

```diff
@@ -635,6 +637,59 @@ FROM [Posts] AS [p]
 WHERE [p].[Title] = @__p_0 AND [p].[Id] = @__id_1
 ```
 
+<a name="parameterized-collections"></a>
+
+#### Parameterized primitive collections
+
+EF8 changed the way [some queries that use primitive collections are translated](xref:core/what-is-new/ef-core-8.0/whatsnew#queries-with-primitive-collections). When a LINQ query contains a parameterized primitive collection, EF will try to convert its contents to JSON and pass it as a single parameter value the query.
+
+<!--
+#region DefaultParameterizationPrimitiveCollection
+async Task<List<Post>> GetPostsPrimitiveCollection(int[] ids)
+    => await context.Posts
+        .Where(e => e.Title == ".NET Blog" && ids.Contains(e.Id))
+        .ToListAsync();
+-->
+[!code-csharp[ForceParameter](../../../../samples/core/Miscellaneous/NewInEFCore9/QuerySample.cs?name=DefaultParameterizationPrimitiveCollection)]
+
+will result in the following translation on SQL Server:
+
+```output
+Executed DbCommand (5ms) [Parameters=[@__ids_0='[1,2,3]' (Size = 4000)], CommandType='Text', CommandTimeout='30']
+SELECT [p].[Id], [p].[Archived], [p].[AuthorId], [p].[BlogId], [p].[Content], [p].[Discriminator], [p].[PublishedOn], [p].[Rating], [p].[Title], [p].[PromoText], [p].[Metadata]
+FROM [Posts] AS [p]
+WHERE [p].[Title] = N'.NET Blog' AND [p].[Id] IN (
+    SELECT [i].[value]
+    FROM OPENJSON(@__ids_0) WITH ([value] int '$') AS [i]
+)
+```
+
+This allows having the same SQL query for different parameterized collections (only the parameter value changes), but in some situations it can lead to performance issues as the database isn't able to optimally plan for the query. `EF.Constant` method can be used to revert to the previous translation.
```

Comment:
```suggestion
This allows having the same SQL query for different parameterized collections (only the parameter value changes), but in some situations it can lead to performance issues as the database isn't able to optimally plan for the query. The `EF.Constant` method can be used to revert to the previous translation.
```